src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetFeedbackDiagnosticFileProvider.cs

```diff
@@ -1,24 +1,110 @@
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
 using Microsoft.Internal.VisualStudio.Shell.Embeddable.Feedback;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
 using NuGet.PackageManagement;
+using NuGet.VisualStudio.Telemetry;
+
+#nullable enable
 
 namespace NuGet.VisualStudio.Common
 {
     [Export(typeof(IFeedbackDiagnosticFileProvider))]
     public class NuGetFeedbackDiagnosticFileProvider : IFeedbackDiagnosticFileProvider
     {
+        // Don't crash VS feedback if imports fail. Maybe broken MEF is exactly what the customer is trying to report!
+        [Import(AllowDefault = true)]
+        public ISolutionManager? SolutionManager { get; set; }
+
         public IReadOnlyCollection<string> GetFiles()
         {
-            // todo: https://github.com/NuGet/Home/issues/8605
+            string filePath = GetFilePath();
+
+            // See comments on IFeedbackDiagnosticFileProvider.GetFiles
+            ThreadHelper.JoinableTaskFactory.RunAsync(() => WriteToZip(filePath))
+                .PostOnFailure(nameof(NuGetFeedbackDiagnosticFileProvider), nameof(WriteToZip));
 
             return new List<string>()
             {
-                DependencyGraphRestoreUtility.GetDefaultDGSpecFileName()
+                filePath
             };
         }
+
+        public async Task WriteToZip(string filePath)
+        {
+            var telemetry = new TelemetryEvent("feedback");
+            telemetry["IsDebuggerAttached"] = Debugger.IsAttached;
+            bool successful = false;
+            Stopwatch sw = Stopwatch.StartNew();
+            try
+            {
+
+                using (var fileStream = File.Open(filePath, FileMode.Create, FileAccess.Write, FileShare.None))
+                using (var zip = new ZipArchive(fileStream, ZipArchiveMode.Create))
+                {
+                    await AddDgSpecAsync(zip);
+                }
+
+                successful = true;
+            }
+            finally
+            {
+                sw.Stop();
+                telemetry["successful"] = successful;
+                telemetry["duration_ms"] = sw.Elapsed.TotalMilliseconds;
+                TelemetryActivity.EmitTelemetryEvent(telemetry);
+            }
+        }
+
+        private string GetFilePath()
+        {
+            string fileNamePrefix = "nuget." + DateTime.UtcNow.ToString("yyyy-MM-dd.HH-mm-ss");
+            int attempts = 0;
+            for (; ; )
+            {
+                attempts++;
+                var fileName = attempts == 0 ? fileNamePrefix + ".zip" : fileNamePrefix + "." + attempts + ".zip";
+                var fullPath = Path.Combine(Path.GetTempPath(), fileName);
+                if (!File.Exists(fullPath))
+                {
+                    return fullPath;
+                }
+            }
+        }
+
+        private async Task AddDgSpecAsync(ZipArchive zip)
+        {
+            string? tempFile = null;
+            try
```

Comment:
I would wrap the try catch body in a JTF.RunAsync({ ... } ).PostOnFailure to avoid catching general exceptions.

Reply:
I disagree.

Firstly, JTF is a tool that has two features:
1. It lets non-async code block the UI thread while waiting for async code to complete, and if that async code happens to need the UI thread, it doesn't deadlock. But it prevents other async code from using the UI thread.
2. It allows async code to block VS shutdown so it can gracefully stop, rather than being killed in a non-deterministic location.

So to me, JTF feels like the wrong tool for the job (log unhandled exceptions).

Secondly, `PostOnFailure` makes it "fire and forget", which means that the calling method can't wait for the async work to finish. This method, `AddDgSpecAsync`, needs to write to the zip archive, but if the method returns to its caller before the fire-and-forget code writes to the zip file, then by the time the fire-and-forget code runs, the zip file will already be closed/disposed. Therefore, I don't see how the suggestion can possibly work.

Reply:
I din't know that JTF is an special construct to work with the UI thread. I just want to avoid having a general Exception catching block in this try-catch block.